iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp

```diff
@@ -740,9 +747,60 @@ struct RemoveLinalgRange : public OpConversionPattern<linalg::RangeOp> {
 };
 }  // namespace
 
-void populateParallelLoopToWorkgroupPatterns(
+static LogicalResult linalgCopyToVector(linalg::CopyOp copyOp,
+                                        ArrayRef<Value> operands,
+                                        ConversionPatternRewriter &rewriter) {
+  linalg::LinalgTilingOptions options;
+  // Tile to memory access of 128bits as those tend to be optimal on most GPUs.
+  constexpr unsigned vecLoadSize = 128;
+  unsigned elementSize = copyOp.getSource()
+                             .getType()
+                             .cast<MemRefType>()
+                             .getElementType()
+                             .getIntOrFloatBitWidth();
+  if (elementSize == 0 || vecLoadSize % elementSize != 0) return failure();
+  unsigned numElement = vecLoadSize / elementSize;
+  options.setTileSizes({1, numElement})
+      .setLoopType(linalg::LinalgTilingLoopType::ParallelLoops);
+  Optional<linalg::TiledLinalgOp> tiledOp = linalg::tileLinalgOp(
+      rewriter, cast<linalg::LinalgOp>(copyOp.getOperation()), options);
+  if (!tiledOp) return failure();
+  if (tiledOp->loops.empty()) return success();
+  setMarker(tiledOp->op, getVectorizeMarker());
+  scf::ParallelOp pLoopOp = dyn_cast<scf::ParallelOp>(tiledOp->loops[0]);
```

Comment:
I think we don't expect the failure on casting here right? So maybe just use `cast` instead.

By the way, we can use `auto` here because the `cast` already states the type clearly. This is optional.